lms/views/api/grading.py

```diff
@@ -117,6 +118,11 @@ def record_canvas_speedgrader_submission(self):
             )
 
         except ExternalRequestError as err:
+            if err.is_timeout:
+                # We'll inform the frontend that this is a retryable error for timeouts.
+                err.retryable = True
+                raise
```

Comment:
I reckon none of the following conditions would match for a timeout, so you could just set `err.retryable = err.is_timeout`, but this is definitely more predictable.